src/utils/exec.c

```diff
@@ -287,16 +288,26 @@ gboolean bd_utils_exec_and_report_status_error (const gchar **argv, const BDExtr
  */
 gboolean bd_utils_exec_and_capture_output (const gchar **argv, const BDExtraArg **extra, gchar **output, GError **error) {
```

Comment:
I wonder if we could just call `bd_utils_exec_and_report_progress()` with progress reporting arguments being `NULL`. Because re-implementing `g_spawn_sync()` properly would be quite a bit harder -- the child process may not terminate if the pipe buffers are not big enough for its output and the parent process is not reading from the pipes.

Reply:
Fixed, `bd_utils_exec_and_capture_output` now shares code with `bd_utils_exec_and_report_progress`.